elements/pfe-accordion/BaseAccordionHeader.ts

```diff
@@ -36,140 +34,102 @@ export abstract class BaseAccordionHeader extends LitElement {
 
   static override readonly shadowRootOptions = { ...LitElement.shadowRootOptions, delegatesFocus: true };
 
-  @property({ attribute: 'aria-controls', reflect: true }) ariaControls?: string;
-
-  @property({ reflect: true }) bordered?: 'true'|'false';
-
-  @property({ reflect: true }) icon?: string;
-
-  @property({ reflect: true, attribute: 'icon-set' }) iconSet?: string;
-
-  @observed
   @property({ type: Boolean, reflect: true }) expanded = false;
 
   @property({ reflect: true, attribute: 'heading-text' }) headingText = '';
 
   @property({ reflect: true, attribute: 'heading-tag' }) headingTag = 'h3';
 
-  /**
-   * Sets color theme based on parent context
-   */
-  @colorContextConsumer()
-  @property({ reflect: true }) on?: ColorTheme;
-
-  @query('.toggle') button?: HTMLButtonElement;
-
-  private _generatedHtag?: HTMLHeadingElement;
+  #generatedHtag?: HTMLHeadingElement;
 
-  private slots = new SlotController(this, 'accents', null);
-
-  private logger = new Logger(this);
-
-  private get ariaExpandedState(): 'true'|'false' {
-    return String(!!this.expanded) as 'true'|'false';
-  }
-
-  constructor() {
-    super();
-    this.addEventListener('click', this._clickHandler);
-  }
+  #logger = new Logger(this);
 
   override connectedCallback() {
     super.connectedCallback();
+    this.addEventListener('click', this.#onClick);
     this.hidden = true;
-    this.id ||= getRandomId('pfe-accordion');
+    this.id ||= getRandomId(this.localName);
+    this.#initHeader();
   }
 
-  @initializer() protected async _init() {
-    const header = await this._getHeaderElement();
+  async #initHeader() {
+    const header = this.#getOrCreateHeader();
 
     // prevent double-logging
-    if (header !== this._generatedHtag) {
-      this._generatedHtag = undefined;
+    if (header !== this.#generatedHtag) {
+      this.#generatedHtag = undefined;
     }
 
     this.headingTag = header?.tagName.toLowerCase() ?? 'h3';
     this.headingText = header?.textContent?.trim() ?? '';
 
+    do {
+      await this.updateComplete;
```

Comment:
What we're doing here is checking to see if `updateComplete` resolves as `false`, which indicates that another update was triggered during the previous update cycle. In which case, we want to wait on the now-enqueued next update. We should do that because we overrode `getUpdateComplete` to include the child pairs' update cycles as well